pennylane/templates/subroutines/qubitization.py

```diff
@@ -150,7 +150,11 @@ def compute_decomposition(*_, **kwargs):  # pylint: disable=arguments-differ
 
         decomp_ops = []
 
-        decomp_ops.append(qml.Reflection(qml.Identity(control)))
+        # QJIT is not compatible with qml.Identity(control), so we build the
+        # identity as a tensor product instead
+        identity = qml.prod(*[qml.Identity(wire) for wire in control])
```

Comment:
Do we need to do this always or only when used in a `QJIT` setting?

Reply:
It's only needed for QJIT. Doing something like `qml.Identity(list_of_wires)` is not supported, so the identity operator needs to be built another way.